File: bms/go/go-gt/env_update/env_update.go
package test

import (
	"math"
)

var ()

const ()

func (e *Env) update() {
	var i int64
	var d int64 = math.MaxInt64
	for i = 0; i < e.N; i++ {
		if !e.T[i] {
			d = min(d, e.Slack[i])
		}
	}
	for i = 0; i < e.N; i++ {
		if e.S[i] {
			e.Lx[i] -= d
		}
	}
	for i = 0; i < e.N; i++ {
		if e.T[i] {
			e.Ly[i] += d
		}
	}
	for i = 0; i < e.N; i++ {
		if !e.T[i] {
			e.Slack[i] -= d
		}
	}
}

func min(a, b int64) int64 {
	if a < b {
		return a
	}
	return b
}

type Env struct {
	N int64

	T, S  []bool
	Slack []int64

	Lx, Ly []int64
}

File: translations_only/success/go/go-gt/env_update/gpt4/Hinted/env_update.rs

struct Env {
    n: i64,
    t: Vec<bool>,
    s: Vec<bool>,
    slack: Vec<i64>,
    lx: Vec<i64>,
    ly: Vec<i64>,
}

impl Env {
    fn update(&mut self) {
        let mut d = i64::MAX;
        for i in 0..self.n {
            if !self.t[i as usize] {
                d = min(d, self.slack[i as usize]);
            }
        }
        for i in 0..self.n {
            if self.s[i as usize] {
                self.lx[i as usize] -= d;
            }
        }
        for i in 0..self.n {
            if self.t[i as usize] {
                self.ly[i as usize] += d;
            }
        }
        for i in 0..self.n {
            if !self.t[i as usize] {
                self.slack[i as usize] -= d;
            }
        }
    }
}

fn min(a: i64, b: i64) -> i64 {
    if a < b { a } else { b }
}
